File: Python2/Arcade/Intro/matrixElementsSum.py
def matrixElementsSum(matrix):
    for i in xrange(len(matrix)-1):
        for j in xrange(len(matrix[i])):
            if matrix[i][j] == 0:
                for k in xrange(i, len(matrix)):
                    matrix[k][j] = 0

    return sum(sum(matrix, []))

File: R/Arcade/Intro/matrixElementsSum.R
matrixElementsSum <- function(m) {
  m <- matrix(unlist(m), nrow = length(m), byrow = TRUE)
  set0 <- function(x) if (any(x == 0)) c(x[1:which.max(x == 0)], rep(0, length(x) - which.max(x == 0))) else x
  sum(apply(m, 2, set0))
}
